test/integration_gcp_test.go

```diff
@@ -130,6 +133,80 @@ func TestGcpParallelStateInit(t *testing.T) {
 	runTerragrunt(t, "terragrunt run-all apply -auto-approve --terragrunt-non-interactive --terragrunt-working-dir "+tmpEnvPath)
 }
 
+func TestGcpOutputFromRemoteState(t *testing.T) { //nolint: paralleltest
```

Comment:
```
=== RUN   TestGcpOutputFromRemoteState
    integration_gcp_test.go:146: Copying fixtures/gcs-output-from-remote-state to /tmp/terragrunt-test3565337247
    integration_gcp_test.go:149: Error writing temp file to /tmp/terragrunt-test3565337247: open /tmp/terragrunt-test3565337247: is a directory
    panic.go:626: Deleting test GCS bucket terragrunt-test-bucket-yzmacx
    panic.go:626: Failed to list objects and versions in GCS bucket terragrunt-test-bucket-yzmacx: storage: bucket doesn't exist
--- FAIL: TestGcpOutputFromRemoteState (0.14s)

```

Reply:
Thanks for running the tests, something was wrong indeed. I fixed this line, hopefully the test should pass now 🤞

Reply:
well... it still fails

```
=== RUN   TestGcpMockOutputsFromRemoteState
    integration_gcp_test.go:189: Copying fixtures/gcs-output-from-remote-state to /tmp/terragrunt-test3189224221
    integration_gcp_test.go:197: [terragrunt apply --terragrunt-fetch-dependency-output-from-state --auto-approve --terragrunt-non-interactive --terragrunt-working-dir /tmp/terragrunt-test3189224221/fixtures/gcs-output-from-remote-state/env1/app1 --terragrunt-disable-log-formatting]
time=2024-09-20T17:51:09Z level=error prefix=. msg=Error: Unsupported block type


time=2024-09-20T17:51:09Z level=error prefix=. msg=  on ../../terragrunt.hcl line 4, in remote_state:

time=2024-09-20T17:51:09Z level=error prefix=. msg=   4:   generate {

time=2024-09-20T17:51:09Z level=error prefix=. msg=

time=2024-09-20T17:51:09Z level=error prefix=. msg=Blocks of type "generate" are not expected here. Did you mean to define argument
"generate"? If so, use the equals sign to assign it a value.


    integration_gcp_test.go:197: Failed to run Terragrunt command 'terragrunt apply --terragrunt-fetch-dependency-output-from-state --auto-approve --terragrunt-non-interactive --terragrunt-working-dir /tmp/terragrunt-test3189224221/fixtures/gcs-output-from-remote-state/env1/app1' due to error: hcl.Diagnostics /tmp/terragrunt-test3189224221/fixtures/gcs-output-from-remote-state/terragrunt.hcl:4,3-11: Unsupported block type; Blocks of type "generate" are not expected here. Did you mean to define argument "generate"? If so, use the equals sign to assign it a value.
        /home/circleci/project/config/hclparse/file.go:70 (0x10077ac)
        	(*File).Decode: return errors.WithStackTrace(err)
        /home/circleci/project/config/config.go:937 (0x1609b3f)
        	decodeAsTerragruntConfigFile: if err := file.Decode(&terragruntConfig, evalContext); err != nil {
        /home/circleci/project/config/config.go:864 (0x16092af)
        	ParseConfig: terragruntConfigFile, err := decodeAsTerragruntConfigFile(ctx, file, evalContext)
        /home/circleci/project/config/config.go:773 (0x1608eb8)
        	ParseConfigFile.func1: config, err = ParseConfig(ctx, file, includeFromChild) //nolint:contextcheck
        /home/circleci/project/telemetry/metrics.go:42 (0x10a0793)
        	Time: return fn(ctx)
        /home/circleci/project/telemetry/telemetry.go:80 (0x1608aad)
        	ParseConfigFile.Telemetry.func2: return Time(ctx, name, attrs, fn)
        /home/circleci/project/telemetry/traces.go:38 (0x10a30dd)
        	Trace: return fn(ctx)
        /home/circleci/project/telemetry/telemetry.go:79 (0x1608a25)
        	ParseConfigFile: return Trace(ctx, name, attrs, func(ctx context.Context) error {
        /home/circleci/project/config/config.go:730 (0x16087f3)
        	ParseConfigFile: err := telemetry.Telemetry(ctx, ctx.TerragruntOptions, "parse_config_file", map[string]interface{}{
        /home/circleci/project/config/include.go:96 (0x16276f7)
        	parseIncludedConfig: return ParseConfigFile(ctx, includePath, includedConfig)
        /home/circleci/project/config/include.go:128 (0x162790a)
        	handleInclude: parsedIncludeConfig, err = parseIncludedConfig(ctx, &includeConfig)
        /home/circleci/project/config/config.go:880 (0x1609310)
        	ParseConfig: mergedConfig, err := handleInclude(ctx, config, false)
        /home/circleci/project/config/config.go:773 (0x1608eb8)
        	ParseConfigFile.func1: config, err = ParseConfig(ctx, file, includeFromChild) //nolint:contextcheck
        /home/circleci/project/telemetry/metrics.go:42 (0x10a0793)
        	Time: return fn(ctx)
        /home/circleci/project/telemetry/telemetry.go:80 (0x1608aad)
        	ParseConfigFile.Telemetry.func2: return Time(ctx, name, attrs, fn)
        /home/circleci/project/telemetry/traces.go:38 (0x10a30dd)
        	Trace: return fn(ctx)
        /home/circleci/project/telemetry/telemetry.go:79 (0x1608a25)
        	ParseConfigFile: return Trace(ctx, name, attrs, func(ctx context.Context) error {
        /home/circleci/project/config/config.go:730 (0x16087f3)
        	ParseConfigFile: err := telemetry.Telemetry(ctx, ctx.TerragruntOptions, "parse_config_file", map[string]interface{}{
        /home/circleci/project/config/config.go:720 (0x160871b)
        	ReadTerragruntConfig: return ParseConfigFile(parcingCtx, terragruntOptions.TerragruntConfigPath, nil) //nolint:contextcheck
        /home/circleci/project/cli/commands/terraform/action.go:101 (0x184dcc5)
        	runTerraform: terragruntConfig, err := config.ReadTerragruntConfig(ctx, terragruntOptions, config.DefaultParserOptions(terragruntOptions))
        /home/circleci/project/cli/commands/terraform/action.go:83 (0x184dabc)
        	Run: return runTerraform(ctx, opts, new(Target))
        /home/circleci/project/cli/commands/terraform/command.go:47 (0x1e48a4d)
        	NewApp.NewCommand.Action.func5: return Run(ctx.Context, opts.OptionsFromContext(ctx))
        /home/circleci/project/cli/app.go:239 (0x1e4a647)
        	runAction.func2: return action(cliCtx)
        /home/circleci/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 (0x1603036)
        	(*Group).Go.func1: if err := f(); err != nil {
        /usr/local/go/src/runtime/asm_amd64.s:1695 (0x47aae1)
        	goexit: BYTE	$0x90	// NOP
        
        
        Stdout: (see log output above)
        
        Stderr: (see log output above)
    panic.go:626: Deleting test GCS bucket terragrunt-test-bucket-3f2cvv
    panic.go:626: Failed to list objects and versions in GCS bucket terragrunt-test-bucket-3f2cvv: storage: bucket doesn't exist
--- FAIL: TestGcpMockOutputsFromRemoteState (0.22s)
```

Reply:
Updated the test again, it should be OK new